src/xSTIR/pSTIR/STIR.py

```diff
@@ -1049,7 +1124,56 @@ def backward(self, ad, subset_num = 0, num_subsets = 1):
             (self.handle, ad.handle, subset_num, num_subsets)
         check_status(image.handle)
         return image
+    def get_linear_acquisition_model(self):
+        '''
+        Returns the linear part of the AcquisitionModel
+        '''
+        am = type(self)()
+        am.set_up( self.acq_templ, self.img_templ )
+        return am
+    def direct(self, image, subset_num = 0, num_subsets = 1, ad = None):
+        '''Projects an image into the (simulated) acquisition space,
+           if the AcquisitionModel is linear.
 
+           Added for CCPi CIL compatibility
+           https://github.com/CCPPETMR/SIRF/pull/237#issuecomment-439894266
+        '''
+        if self.is_linear():
+            return self.forward(image, \
+                                subset_num=subset_num, \
+                                num_subsets = num_subsets, \
+                                ad = ad)
+        else:
+            raise error('AcquisitionModel is not linear\nYou can get the linear part of the AcquisitionModel with get_linear_acquisition_model')
+    def adjoint(self, ad, subset_num = 0, num_subsets = 1):
+        '''Back-projects acquisition data into image space, if the
+           AcquisitionModel is linear
+
+           Added for CCPi CIL compatibility
+           https://github.com/CCPPETMR/SIRF/pull/237#issuecomment-439894266
+        '''
+        if self.is_linear():
+            return self.backward(ad, subset_num = subset_num, 
+                             num_subsets = num_subsets)
+        else:
+            raise error('AcquisitionModel is not linear')
+    def is_affine(self):
+        '''Returns whether the AcquisitionModel is affine'''
```

Comment:
see suggested change for MR.
possibly we should add in both that it is currently always True